i6_models/parts/frontend/generic_frontend.py

```diff
@@ -0,0 +1,224 @@
+from __future__ import annotations
+
+__all__ = [
+    "FrontendLayerType",
+    "GenericFrontendV1Config",
+    "GenericFrontendV1",
+]
+
+from dataclasses import dataclass
+from enum import Enum, auto
+from typing import Callable, Optional, Tuple, Union, Sequence
+
+import torch
+from torch import nn
+
+from i6_models.config import ModelConfiguration
+
+from i6_models.parts.frontend.common import get_same_padding, mask_pool, calculate_output_dim
+
+
+class FrontendLayerType(Enum):
+    Conv2d = auto()
+    Pool2d = auto()
+    Activation = auto()
+
+
+@dataclass
+class GenericFrontendV1Config(ModelConfiguration):
+    """
+    Attributes:
+        in_features: number of input features to module
+        layer_ordering: the ordering of the front end layer sequences, the ordering element must be selected from FrontendLayerType
+            e.g. the ordering of VGG4LayerActFrontendV1 would be [FrontendLayerType.Conv2d, FrontendLayerType.Activation,
+            FrontendLayerType.Pool2d, FrontendLayerType.Conv2d, FrontendLayerType.Conv2d, FrontendLayerType.Activation,
+            FrontendLayerType.Pool2d]
+        conv_kernel_sizes: kernel sizes for each conv layer
+        conv_strides: stride sizes for each conv layer
+        conv_paddings: paddings sizes for each conv layer
+        conv_out_dims: number of out channels for each conv layer
+        pool_kernel_sizes: kernel sizes for each pool layer
+        pool_strides: stride sizes for each pool layer
+        pool_paddings: padding sizes for each pool layer
+        activations: activation functions
+        out_features: output size of the final linear layer
+    """
+
+    in_features: int
+    layer_ordering: Sequence[FrontendLayerType]
+    conv_kernel_sizes: Optional[Sequence[Union[int, Tuple[int, int]]]]
+    conv_strides: Optional[Sequence[Union[int, Tuple[int, int]]]]
+    conv_paddings: Optional[Sequence[Union[int, Tuple[int, int]]]]
+    conv_out_dims: Optional[Sequence[Union[int, Tuple[int, int]]]]
+    pool_kernel_sizes: Optional[Sequence[Union[int, Tuple[int, int]]]]
+    pool_strides: Optional[Sequence[Union[int, Tuple[int, int]]]]
+    pool_paddings: Optional[Sequence[Union[int, Tuple[int, int]]]]
```

Comment:
```suggestion
    conv_kernel_sizes: Optional[Sequence[Tuple[int, int]]]
    conv_strides: Optional[Sequence[Tuple[int, int]]]
    conv_paddings: Optional[Sequence[Tuple[int, int]]]
    conv_out_dims: Optional[Sequence[int]]
    pool_kernel_sizes: Optional[Sequence[Tuple[int, int]]]
    pool_strides: Optional[Sequence[Tuple[int, int]]]
    pool_paddings: Optional[Sequence[Tuple[int, int]]]
```
In the previous frontend implementation under `i6_models/parts/frontend/vgg_act.py` we only allowed `Tuple[int, int]` and not just `int`. I would personally prefer to keep it that way because it's more explicit and also slightly simplifies the code.

The exception is `conv_out_dims` which was an `int` before but now is mistakenly allowed to be a tuple.